### ISSUE TITLE
assert-dev-server-stopped.py never matches the app's own Electron process, so the guard and the reaper disagree

### ISSUE THREAD

**Issue by dzackgarza** [dzackgarza/zettlr-pandoc#56]
## Observed behavior

`scripts/assert-dev-server-stopped.py --kill` reports terminating only the `electron-forge` node process and leaves the app's own Electron process running:

```
$ python3 scripts/assert-dev-server-stopped.py --kill
Freeing dev ports: terminating stale dev process(es) [784861]
$ pgrep -af "Zettlr-pandoc-mathjax.*electron"
787325 /home/dzack/gitclones/Zettlr-pandoc-mathjax/node_modules/electron/dist/electron . /tmp/probe.md
```

The surviving process had to be killed by hand before `just test-e2e` would run.

## Mechanism

`dev_pids()` builds `command` by splitting `/proc/<pid>/cmdline` on NUL, then tests:

```python
is_project_electron = any(token.endswith("node_modules/electron/dist/electron") for token in command) and "." in command
```

Electron's main process does not present argv as NUL-separated tokens the way the node processes do, so `command` is not the token list this predicate assumes: no element ends with `node_modules/electron/dist/electron`, and `"."` is not an element. Both halves of the conjunction fail, so the app's own process is never matched. The `is_forge_start` branch still matches the node processes, which is why the guard appears to work.

## Why it matters beyond the reaper

`dev_server_is_running()` masks the gap:

```python
if dev_pids():
    return True
# falls through to a port probe on 127.0.0.1:3100
```

The port probe catches a live dev server, so the *guard* usually still refuses correctly. But `kill_dev_processes()` has no such fallback — it acts only on `dev_pids()`. So the two functions disagree about what is running: the guard says "something is up", the reaper says "nothing to kill", and the caller is left in a loop it cannot clear.

A dev instance can therefore survive an entire `just test-e2e` run while the guard reports the tree clean, which is worth more than the reaper bug: e2e specs launch their own Electron against a repo-wide `.webpack/` build directory, so an unnoticed dev instance is free to corrupt a run.

## Expected behavior

`dev_pids()` identifies the project's Electron main process regardless of how its argv is exposed in `/proc`, so the guard and the reaper agree on the same process set. A reasonable check is the resolved `/proc/<pid>/exe` against `node_modules/electron/dist/electron` together with the existing `cwd == PROJECT_ROOT` test, rather than parsing `cmdline`.

Verification should assert against a real launched dev instance, since the defect is precisely that the synthetic shape and the real one differ.

**Comment by dzackgarza** [comment]
Fixed on develop by ec257611c: the reaper now matches the app's Electron process by its resolved executable, so guard and reaper agree.
